File: Data Structures & Algorithms/concatenation-of-array/submission-2.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int length = nums.size();
        vector<int> ans;
        for(int i = 0; i < length*2; i++)
        {
            ans.push_back(nums[i%length]);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/binary-search/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
    
        return binarySearch(nums, target, 0, nums.length);
    }

    public int binarySearch(int[] nums, int target, int left, int right){
        int index = (left+right)/2;
        if(left  == right && nums[index] != target){
            return -1;
        }
        if(nums[index] == target){
            return index;
        } 
        if(right - left == 1){
            return -1;
        }
        if(target > nums[index]){
            return binarySearch(nums, target, index, right);
        } else{
            return binarySearch(nums, target, left, index);
        }


    }


}
